### ISSUE TITLE
Submit a game "Get Info" functionality

### ISSUE THREAD

**Issue by foohonpie** [foohonpie/pbbg.com#13]
The initial sate of the "Submit a Game" form should be a single text input for a URL with a "Get Info" button.

When a url is submitted, the following should happen:

1. URL syntax validated
  a. If error, display error message
  b. if valid, continue
2. Submit url to TBD API endpoint
  a. If error, display error message
  b. if valid, populate fields with response data

Step #2 is blocked by an API dependency.  We can fake a success response in the short term and leave the other fields bank.

**Comment by bpkennedy** [comment]
I am assuming that the TBD api is going to return a string for me to pop into the input?

otherwise, for the Click of the 'Get Info' button I'll show a brief loader inside the button itself while the simulated request is made.

**Comment by foohonpie** [comment]
it'll likely return an array of:

1. canonical url (in case it's different than what was submitted)
2. meta title
3. meta description
4. url to homepage screenshot image

**Comment by foohonpie** [comment]
From https://github.com/pbbg/pbbg.com/pull/21#issuecomment-711092323:

> There's an issue when re-submitting after a URL error. Steps to reproduce:
> 
> 1. intentionally enter a pre-existing url
> 2. click "save" and get the error message
> 3. change the value in the URL field to a new url, but don't click "get info"
> 4. click "save" and get the same error message
> 
> "Save" retains the original value, instead of sending over the updated URL value.

The issue is that the single URL input is doing double-duty: first calling `GET /info` and then also submitting to `POST /games`.  We have some options:

1. have the URL input update the request body going to `POST /games` when the user changes its value
  * This raises a new issue: when the user change this value, we have no way to know which form they intend to re-submit to.  For example, if they type in a new value and hit enter, we would inevitably do the wrong thing for some users.
2. if the user focuses on the URL field after the `GET /info` call, reset the form and start them over.
  * This is "clean" and clear, but disruptive to the flow.
3. have 2 different form inputs
  * This will look a little noisier, but removes the ambiguity.
  * This also will help handle the case when a URL submitted to `GET /info` returns a canonical URL that is different.

**Comment by foohonpie** [comment]
From https://github.com/pbbg/pbbg.com/pull/21#issuecomment-711092323:

> There's an issue when re-submitting after a URL error. Steps to reproduce:
> 
> 1. intentionally enter a pre-existing url
> 2. click "save" and get the error message
> 3. change the value in the URL field to a new url, but don't click "get info"
> 4. click "save" and get the same error message
> 
> "Save" retains the original value, instead of sending over the updated URL value.

The issue is that the single URL input is presented to the user as a field for 2 different forms: first to `GET /info` and then also to `POST /games`.

  We have some options:

1. have the URL input update the request body going to `POST /games` when the user changes its value
   * This raises a new issue: when the user change this value, we have no way to know which form they intend to re-submit to. 
 If they type in a new value and hit enter, we would inevitably do the wrong thing for some users.

2. if the user focuses on the URL field after the `GET /info` call, reset the form and start them over at `GET /info`.
   * This is "clean" and clear, but disruptive to the flow.  This adds friction and seems annoying if all I wanted to do was update the URL and submit.

3. from Discord:

   > the api should return a 422 or something saying that the url already exists.  At that point, we can show the error toast message saying the url has been used already instead of revealing the rest of the form fields.
And when the error happens, the rest of the form won't be there.  They will have to change the url and click Get Info again to continue.

    * This is similar to `#3` above, but only happens upon error.  This also puts the onus on `GET /info` to make a determination on the validity of the URL, vs just returning information about a website.

4. have 2 different form inputs.
   * This will look a little noisier, but removes the ambiguity.
   * This also will help handle the case when a URL submitted to `GET /info` returns a canonical URL that is different.

<img width="925" alt="Screen Shot 2020-10-18 at 9 29 35 AM" src="https://user-images.githubusercontent.com/674157/96373947-899a0780-1124-11eb-9ff2-0f83971dc91b.png">

5. Remove the `GET /info` button/form after submission
   * This will make `GET /info` a "step 1" that gets out of your way after you do it.  We would need to give the user a way to walk back to this step, as it wouldn't live alongside the `POST /games` form anymore.
  
These all have some tradeoffs, but I think we should try not to be too clever as doing so requires assumptions on what we think the user wants to do.  The desired flow that tripped this error in the first place was simple:

> I want to update the URL that will be submitted, and that's it.  That's all I want to change.

Thoughts?

**Comment by foohonpie** [comment]
This above discussion and use case is outside of the scope of the rewrite.  Let's revisit it after launch.
